### PR TITLE
converting models to CSV

### PR DESCRIPTION
**Pull request by dyf**
1. Turned all non-Enum models into CSVs
2. Added utils.py which dynamically generates models
3. renamed registry.py -> registries.py
4. standardized singular/plural model class names
5. removed the "screaming snake case" class variable names - now just title case

The reason for 5 is that I think it's confusing to add what's basically a third property to each class instance that is inconsistent with the `name` and `abbreviation`. 

This is currently backwards incompatible.

### CODE REVIEW

File: src/aind_data_schema_models/modalities.py [hunk 1]
```diff
@@ -1,144 +1,33 @@
...
-    abbreviation: Literal["slap"] = "slap"
-
-
-class Spim(_Modality):
-    """Spim"""
-
-    name: Literal["Selective plane illumination microscopy"] = "Selective plane illumination microscopy"
-    abbreviation: Literal["SPIM"] = "SPIM"
-
+Modalities = create_literal_class(
```
Comment (jtyoung84) [inline src/aind_data_schema_models/modalities.py:17]: Renaming this is going to be a pain

File: src/aind_data_schema_models/modalities.py [hunk 2]
```diff
@@ -1,144 +1,33 @@
...
-    FIB = Fib()
-    ISI = Isi()
-    MERFISH = Merfish()
-    MRI = Mri()
-    POPHYS = POphys()
-    SLAP = Slap()
-    SPIM = Spim()
-    _ALL = tuple(_Modality.__subclasses__())
-    ONE_OF = Annotated[Union[_ALL], Field(discriminator="name")]
+@classmethod
```
Comment (jtyoung84) [inline src/aind_data_schema_models/modalities.py:26]: You can remove the decorator and consider making the method private, or putting it into the utils package

File: src/aind_data_schema_models/registries.py [hunk 3]
```diff
@@ -0,0 +1,44 @@
+"""Common registries"""
+
+from typing import Annotated, Union
```
Comment (jtyoung84) [inline src/aind_data_schema_models/registries.py:3]: To make it compatible with python 3.8, should be
`from typing_extensions import Annotated`

**Review by jtyoung84 — CHANGES_REQUESTED** [review]
- Use `from typing_extensions import Annotated`
- Consider keeping the original names the same?
- Remove class decorators

File: src/aind_data_schema_models/modalities.py (hunk 2 above)
Comment (dyf) [reply src/aind_data_schema_models/modalities.py:26]: When I removed the decorator it stopped passing `self` correctly, so I think I still need it.

File: src/aind_data_schema_models/modalities.py (hunk 1 above)
Comment (dyf) [reply src/aind_data_schema_models/modalities.py]: reverted

**Review by saskiad — APPROVED** [review]

**Review by jtyoung84 — APPROVED** [review]
This is a breaking change and will require people to update their code base. We should make sure that the major version is incremented.

**Comment by dyf** [comment]
> This is a breaking change and will require people to update their code base. We should make sure that the major version is incremented.

How do I do this?

**Comment by dyf** [comment]
moved to draft - I started updated the schema to support this and it's a massive change. going to put the screaming snake case class names back.

**Comment by jtyoung84** [comment]
> > This is a breaking change and will require people to update their code base. We should make sure that the major version is incremented.
> 
> How do I do this?

We can put `BREAKING CHANGE` in the commit message. If that doesn't work, then we can manually increment the version.

**Comment by jtyoung84** [comment]
> moved to draft - I started updated the schema to support this and it's a massive change. going to put the screaming snake case class names back.

I made some protected attributes (`_ALL` and `_abbreviation_map` for example) public. I don't think that requires a massive change, but I can revert those if you want this to be backwards compatible.

**Pull request merged**